#921 Avoid generation of duplicate local variable name

sjaakd opened the pull request:
Have a more meaningful name. And add it to the set of existing variables (forgotten initially).

gunnarmorling commented:
@sjaakd Looks good. I'm lacking one piece of info though: in which case would we assign the result of property mappings to local variables instead of passing them to the target getter right away?

gunnarmorling commented:
Merged; thanks, @sjaakd!

sjaakd commented:
> I'm lacking one piece of info though: in which case would we assign the result of property mappings to local variables instead of passing them to the target getter right away?

Hmm. I wish I could state exactly under which conditions. :cry:. It started out with:
1. converted
2. not direct 
3. not mapped.
4. not presence checked.
5. not source type primitive

but some more issues have been written.. So some more conditions are added. You added a condition in #931.. So.. It starts to be a forrest of if-then-else stuff. High in need of refactoring. On the positive side we added test cases for all of them.
